terraform/modules/sfn-io-helper-lambdas/app/sfn_io_helper/stage_io.py

```diff
@@ -121,3 +123,48 @@ def preprocess_sfn_input(sfn_state, aws_region, aws_account_id, state_machine_na
     link_outputs(sfn_state)
 
     return sfn_state
+
+
+def broadcast_stage_complete(execution_id: str, stage: str):
+    if "SQS_QUEUE_URLS" not in os.environ:
+        return
+
+    sqs_queue_urls = os.environ["SQS_QUEUE_URLS"].split(",")
+
+    assert len(execution_id.split(":")) == 8
+    _, _, _, aws_region, aws_account_id, _, state_machine_name, execution_name = execution_id.split(":")
+
+    state_machine_arn = f"arn:aws:states:{aws_region}:{aws_account_id}:stateMachine:{state_machine_name}"
+    execution_arn = f"arn:aws:states:{aws_region}:{aws_account_id}:execution:{execution_id}"
+
+    body = json.dumps({
+        "version": "0",
+        "id": str(uuid4()),
+        "detail-type": "Step Functions Execution Status Change",
+        "source": "aws.states",
+        "account": aws_account_id,
+        "time": datetime.now().strftime('%Y-%m-%dT%H:%M:%SZ'),
+        "region": aws_region,
+        "resources": [execution_arn],
+        "detail": {
+            "executionArn": execution_arn,
+            "stateMachineArn": state_machine_arn,
+            "name": execution_name,
+            "status": "RUNNING",
+            "lastCompletedStage": re.sub(r'(?<!^)(?=[A-Z])', '_', stage).lower(),
+            # "startDate": 1551225271984,
```

Comment:
nit: remove comment?

Reply:
I purposefully left this in to leave a record of the full expected message schema. I can add a comment explaining that